pyhive/sqlalchemy_sparksql.py

```diff
@@ -0,0 +1,72 @@
+from . import sqlalchemy_hive
+
+import re
+
+from sqlalchemy import exc
+from sqlalchemy.engine import default
+
+
+class SparkSqlDialect(sqlalchemy_hive.HiveDialect):
+    name = b'sparksql'
+    execution_ctx_cls = default.DefaultExecutionContext
+
+    def _get_table_columns(self, connection, table_name, schema):
+        full_table = table_name
+        # Only qualify the table if the schema exists and is not default.
+        if schema and schema.lower() != "default":
+            full_table = schema + '.' + table_name
+        # TODO using TGetColumnsReq hangs after sending TFetchResultsReq.
+        # Using DESCRIBE works but is uglier.
+        try:
+            # This needs the table name to be unescaped (no backticks).
+            rows = connection.execute('DESCRIBE {}'.format(full_table)).fetchall()
+        except exc.OperationalError as e:
+            # Does the table exist?
+            regex_fmt = r'TExecuteStatementResp.*NoSuchTableException.*Table or view \'{}\'' \
+                        r' not found'
+            regex = regex_fmt.format(re.escape(table_name))
+            if re.search(regex, e.args[0]):
+                raise exc.NoSuchTableError(full_table)
+            elif schema:
+                schema_regex_fmt = r'TExecuteStatementResp.*NoSuchDatabaseException.*Database ' \
+                                   r'\'{}\' not found'
+                schema_regex = schema_regex_fmt.format(re.escape(schema))
+                if re.search(schema_regex, e.args[0]):
+                    raise exc.NoSuchTableError(full_table)
+            else:
+                # When a hive-only column exists in a table
+                hive_regex_fmt = r'org.apache.spark.SparkException: Cannot recognize hive type ' \
+                                 r'string'
+                if re.search(hive_regex_fmt, e.args[0]):
+                    raise exc.UnreflectableTableError
+                else:
+                    raise
+        else:
+            return rows
+
+    def get_table_names(self, connection, schema=None, **kw):
+        query = 'SHOW TABLES'
+        if schema:
+            query += ' IN ' + self.identifier_preparer.quote_identifier(schema)
+        return list(row[1] for row in filter(
```

Comment:
Using an `if` for comprehensions instead of a filter is usually more Pythonic and easier to read, for example something like:

```python
return [row[1] for row in connection.execute(query) if not row[-1]]
```

Reply:
All the code in this comes from merging this PR: https://github.com/dropbox/PyHive/pull/247/files

Reply:
Ah, Ok, then avoiding conflicts with upstream updates is more important than my tiquismiquismos :)